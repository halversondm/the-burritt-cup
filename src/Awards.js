import React from 'react';
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import Award from './Award';

class Awards extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            awards: [
                {
                    image: "",
                    alt: "",
                    title: "",
                    subtitle: "",
                    metatitle: "",
                    winners: [
                        {
                            year: "",
                            name: ""
                        }
                    ]
                }
            ]
        };
    }

    componentDidMount() {
        fetch('data/awards.json')
            .then(response => response.json())
            .then(json => this.setState({awards: json}));
    }

    render() {
        return (
            <Container>
                <Row className="justify-content-center">
                    <Col lg={8} md={10}>
                        <h1>Awards</h1>
                    </Col>
                </Row>
                {this.state.awards.map((award, i) => {
                    return (
                        <Award key={i} image={award.image} alt={award.alt} title={award.title} subtitle={award.subtitle} metatitle={award.metatitle} winners={award.winners} />
                    );
                })}
            </Container>
        );
    }
}

export default Awards;