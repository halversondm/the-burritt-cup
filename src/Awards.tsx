import {useState, useEffect} from 'react';
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from 'react-bootstrap/Col';
import Award from './Award.js';

export default function Awards() {

    const [state, setState] = useState({
        awards: [
            {
                image: "",
                alt: "",
                title: "",
                subtitle: "",
                metatitle: "",
                winners: [
                    {
                        year: "",
                        name: ""
                    }
                ]
            }
        ]
    });

    useEffect(() => {
        fetch('data/awards.json')
            .then(response => response.json())
            .then(json => setState({awards: json}));
    })


    return (
        <Container>
            <Row className="justify-content-center">
                <Col lg={8} md={10}>
                    <h1>Awards</h1>
                </Col>
            </Row>
            {state.awards.map((award, i) => {
                return (
                    <Award key={i} image={award.image} alt={award.alt} title={award.title} subtitle={award.subtitle}
                           metatitle={award.metatitle} winners={award.winners}/>
                );
            })}
        </Container>
    );
}
