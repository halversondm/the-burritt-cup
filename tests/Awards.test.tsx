import Awards from '../src/Awards';
import {render} from "@testing-library/react";

test("renders without crashing", () => {
  const unit = <Awards />;
  const component = render(unit);
  const tree = component.container;
  expect(tree.toString()).toMatchSnapshot();
});
