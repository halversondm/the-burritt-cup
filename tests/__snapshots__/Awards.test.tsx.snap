// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders without crashing 1`] = `"<div><div class="container"><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><h1>Awards</h1></div></div><div><hr><div class="justify-content-center row"><div class="col-lg-8 col-md-10"><div class="post-preview"><h2 class="post-title"></h2><h4 class="post-subtitle"></h4><p class="post-meta"></p><p class="post-meta">Past Winners:</p><table class="table table-striped table-responsive"><thead><tr><th>Year</th><th>Players</th></tr></thead><tbody><tr><td></td><td></td></tr></tbody></table></div></div></div></div></div></div>"`;
